src/main/io/ledstrip.c

```diff
@@ -943,23 +1076,30 @@ bool parseColor(int index, const char *colorConfig)
 bool setModeColor(ledModeIndex_e modeIndex, int modeColorIndex, int colorIndex)
 {
     // check color
-    if(colorIndex < 0 || colorIndex >= LED_CONFIGURABLE_COLOR_COUNT)
+    if (colorIndex < 0 || colorIndex >= LED_CONFIGURABLE_COLOR_COUNT)
         return false;
-    if(modeIndex < LED_MODE_COUNT) {  // modeIndex_e is unsigned, so one-sided test is enough
+    if (modeIndex < LED_MODE_COUNT) {  // modeIndex_e is unsigned, so one-sided test is enough
         if(modeColorIndex < 0 || modeColorIndex >= LED_DIRECTION_COUNT)
             return false;
         modeColors(modeIndex)->color[modeColorIndex] = colorIndex;
-    } else if(modeIndex == LED_SPECIAL) {
-        if(modeColorIndex < 0 || modeColorIndex >= LED_SPECIAL_COLOR_COUNT)
+    } else if (modeIndex == LED_SPECIAL) {
+        if (modeColorIndex < 0 || modeColorIndex >= LED_SPECIAL_COLOR_COUNT)
             return false;
-        specialColors(0)->color[modeColorIndex] = colorIndex;
+        specialColors()->color[modeColorIndex] = colorIndex;
     } else {
         return false;
     }
     return true;
 }
 
-void pgResetFn_colors(hsvColor_t *instance)
+
+
+void pgResetFn_ledConfigs(ledConfigPacked_t* instance)
+{
+    memcpy_fn(instance, &defaultLedStripConfig, sizeof(defaultLedStripConfig));
+}
+
+void pgResetFn_colors(hsvColor_t* instance)
 {
     // copy hsv colors as default
     BUILD_BUG_ON(ARRAYLEN(*colors_arr()) <= ARRAYLEN(hsv));
```

Comment:
This condition is invalid (will fail if `ARRAYLEN(*colors_arr()) == ARRAYLEN(hsv)`), can you fix it in you PR, please?